lib/application.js

```diff
@@ -296,7 +296,9 @@ app.enableAuth = function() {
     var req = ctx.req;
     var Model = method.ctor;
     var modelInstance = ctx.instance;
-    var modelId = modelInstance && modelInstance.id || req.param('id');
+    var modelId = (modelInstance && modelInstance.id) ||
+      (req.params && req.params.id) ||
+      (req.body && req.body.id);
```

Comment:
`updateOrCreate` is a good point! Though I wonder if a requestor should ever be able to determine a `modelId`, but that is another discussion. (Backwards compatibility is not always a good thing) Let's keep the `req.body` lookup.

@bajtos Do you agree that `req.params.id` should take precedence over `req.body.id`?